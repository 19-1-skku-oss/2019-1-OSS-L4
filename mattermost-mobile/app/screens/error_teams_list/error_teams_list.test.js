// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import FailedNetworkAction from 'app/components/failed_network_action';
import ErrorTeamsList from './error_teams_list';

describe('ErrorTeamsList', () => {
    const navigator = {
        setOnNavigatorEvent: () => {}, // eslint-disable-line no-empty-function
    };

    const loadMe = async () => {
        return {
            data: {},
        };
    };

    const baseProps = {
        actions: {
            loadMe: () => {}, // eslint-disable-line no-empty-function
            connection: () => {}, // eslint-disable-line no-empty-function
            logout: () => {}, // eslint-disable-line no-empty-function
            selectDefaultTeam: () => {}, // eslint-disable-line no-empty-function
        },
        theme: Preferences.THEMES.default,
        navigator,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ErrorTeamsList {...baseProps}/>
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call for userInfo on retry', async () => {
        const connection = jest.fn();
        const selectDefaultTeam = jest.fn();
        const logout = jest.fn();
        const actions = {
            loadMe,
            logout,
            selectDefaultTeam,
            connection,
        };

        const newProps = {
            ...baseProps,
            actions,
        };

        const wrapper = shallow(
            <ErrorTeamsList {...newProps}/>
        );

        wrapper.find(FailedNetworkAction).props().onRetry();
        await loadMe();
        expect(selectDefaultTeam).toHaveBeenCalledTimes(1);
    });
});
