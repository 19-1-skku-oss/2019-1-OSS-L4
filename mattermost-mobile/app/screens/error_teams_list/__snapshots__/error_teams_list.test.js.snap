// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorTeamsList should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <FailedNetworkAction
    errorDescription={
      Object {
        "defaultMessage": "Make sure you have an active connection and try again.",
        "id": "mobile.failed_network_action.shortDescription",
      }
    }
    errorTitle={
      Object {
        "defaultMessage": "Team Not Found",
        "id": "error.team_not_found.title",
      }
    }
    onRetry={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;
